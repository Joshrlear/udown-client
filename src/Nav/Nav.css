/* color scheme */
/* fonts */
* {
  margin: 0;
  padding: 0;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}

body {
  font-family: sans-serif;
}

nav {
  position: fixed;
  display: -ms-grid;
  display: grid;
  -ms-grid-rows: 15% 70% 15%;
      grid-template-rows: 15% 70% 15%;
  -ms-grid-columns: (1fr)[6];
      grid-template-columns: repeat(6, 1fr);
  height: 8vh;
  width: 100vw;
  background: #1c2324;
  z-index: 1;
}

@media screen and (min-width: 900px) {
  nav {
    z-index: 5;
  }
}

.searchbar_container {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  position: relative;
  -ms-grid-column: 1;
      grid-column-start: 1;
  grid-column-end: 5;
  -ms-grid-row: 2;
      grid-row-start: 2;
  grid-row-end: 2;
}

@media screen and (min-width: 1100px) {
  .searchbar_container {
    grid-column-end: 2;
  }
}

.searchbar_input {
  border-radius: 20px;
  width: 35px;
  height: 35px;
  margin-top: 0px;
  margin-bottom: 0px;
  margin-right: 0px;
  margin-left: 20px;
  background-color: rgba(0, 0, 0, 0);
  color: rgba(255, 255, 255, 0);
  padding-left: 15px;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.searchbar_input:focus {
  width: calc(100% - 20px);
  color: rgba(255, 255, 255, 0.6);
}

.searchbar_input::-webkit-input-placeholder {
  color: rgba(255, 255, 255, 0);
}

.searchbar_input:focus::-webkit-input-placeholder {
  color: rgba(255, 255, 255, 0.3);
  text-align: left;
}

#magnifying-glass {
  font-size: 40px;
  /* This controls the size. */
  display: inline-block;
  width: 0.4em;
  height: 0.4em;
  border: 2px solid rgba(255, 255, 255, 0.6);
  position: absolute;
  border-radius: 0.35em;
  margin-left: 28px;
  margin-top: -0.05vw;
  z-index: -1;
  -webkit-transition: all 0.3s ease-out;
  transition: all 0.3s ease-out;
}

#magnifying-glass::before {
  content: "";
  display: inline-block;
  position: absolute;
  right: -6px;
  bottom: -3px;
  border-width: 0;
  background-image: -webkit-gradient(linear, left top, right top, from(rgba(255, 255, 255, 0.3)), to(white));
  background-image: linear-gradient(to right, rgba(255, 255, 255, 0.3), white);
  width: 8px;
  height: 2px;
  -webkit-transform: rotate(45deg);
  -moz-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  -o-transform: rotate(45deg);
  -webkit-transition: all 0.3s ease-out;
  transition: all 0.3s ease-out;
}

#magnifying-glass.off {
  border: none;
}

#magnifying-glass.off::before {
  background-image: -webkit-gradient(linear, left top, right top, from(rgba(255, 255, 255, 0)), to(rgba(255, 255, 255, 0)));
  background-image: linear-gradient(to right, rgba(255, 255, 255, 0), rgba(255, 255, 255, 0));
}

.search {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  position: relative;
  height: 100%;
  margin-left: 0px;
  -ms-grid-column: 6;
      grid-column-start: 6;
  grid-column-end: 6;
  -ms-grid-row: 2;
      grid-row-start: 2;
  grid-row-end: 2;
  z-index: 3;
}

@media screen and (min-width: 1100px) {
  .search {
    -ms-grid-column: 2;
    grid-column: 2/2;
    margin-left: 20px;
  }
}

.search_btn_container {
  background: white;
  opacity: 0;
  position: relative;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  width: 35px;
  height: 35px;
  border-radius: 50%;
  display: -ms-grid;
  display: grid;
  -ms-grid-rows: (1fr)[7];
      grid-template-rows: repeat(7, 1fr);
  -ms-grid-column: 6;
      grid-column-start: 6;
  grid-column-end: 6;
  -ms-grid-row: 4;
      grid-row-start: 4;
  grid-row-end: 5;
  z-index: 2;
  -webkit-transition: opacity 0.35s ease-out;
  transition: opacity 0.35s ease-out;
}

.search_btn_container.on {
  opacity: 1;
}

.search_btn_container .search_btn {
  background: rgba(0, 0, 0, 0);
  height: 3px;
  width: 0px;
  -ms-grid-row: 4;
      grid-row-start: 4;
  grid-row-end: 5;
  z-index: -1;
  margin: 0;
  position: relative;
  cursor: pointer;
  -webkit-transition: all 0.3s ease-out;
  transition: all 0.3s ease-out;
}

.search_btn_container .search_btn:before, .search_btn_container .search_btn:after {
  content: "";
  background: rgba(0, 0, 0, 0);
  position: absolute;
  height: 3px;
  width: 15px;
  -webkit-transition: all 0.3s ease-out;
  transition: all 0.3s ease-out;
}

.search_btn_container .search_btn.on {
  width: 18px;
  border-top-right-radius: 25%;
  border-bottom-right-radius: 25%;
}

.search_btn_container .search_btn.on, .search_btn_container .search_btn.on:before, .search_btn_container .search_btn.on:after {
  margin-left: 6px;
  background: #1c2324;
  z-index: 2;
}

.search_btn_container .search_btn:before {
  right: -6.3px;
  bottom: -4px;
  -webkit-transform: rotate(-45deg);
          transform: rotate(-45deg);
  border-radius: 3px;
}

.search_btn_container .search_btn:after {
  right: -6.3px;
  top: -4px;
  -webkit-transform: rotate(45deg);
          transform: rotate(45deg);
  border-radius: 3px;
}

.newMsg_notification {
  display: none;
}

.newMsg_notification.newMsg {
  display: initial;
  background-color: #c8db16;
  border-radius: 50%;
  width: calc(1.7vw + 5px);
  height: calc(1.7vw + 5px);
  position: relative;
  top: 5%;
  right: 30%;
  margin-left: auto;
  margin-right: auto;
  -ms-grid-column: 6;
      grid-column-start: 6;
  grid-column-end: 7;
  -ms-grid-row: 2;
      grid-row-start: 2;
  grid-row-end: 2;
  pointer-events: none;
  z-index: 4;
}

@media screen and (min-width: 760px) {
  .newMsg_notification.newMsg {
    right: 18%;
  }
}

@media screen and (min-width: 1100px) {
  .newMsg_notification.newMsg {
    display: none;
  }
}

.hamburger_container {
  position: relative;
  left: 90%;
  display: -ms-grid;
  display: grid;
  -ms-grid-rows: (1fr)[7];
      grid-template-rows: repeat(7, 1fr);
  -ms-grid-column: 5;
      grid-column-start: 5;
  grid-column-end: 6;
  -ms-grid-row: 2;
      grid-row-start: 2;
  grid-row-end: 2;
  z-index: 3;
}

@media screen and (min-width: 1100px) {
  .hamburger_container {
    display: none;
  }
}

.hamburger_container.on {
  display: none;
  z-index: -1;
}

.hamburger {
  cursor: pointer;
  -ms-grid-row: 2;
  -ms-grid-row-span: 1;
  grid-row: 2/3;
  -ms-grid-column: 2;
  -ms-grid-column-span: 4;
  grid-column: 2/6;
}

@media screen and (min-width: 760px) {
  .hamburger {
    -ms-grid-column: 2;
    -ms-grid-column-span: 2;
    grid-column: 2/4;
    width: 85%;
  }
}

.line {
  height: 0.5vh;
  background: #e0e0e0;
  margin: 14%;
  -webkit-transition: all 0.3s ease-out;
  transition: all 0.3s ease-out;
}

.line.on {
  width: 0px;
  margin-left: 35px;
  background: transparent;
  z-index: -1;
}

.link.newMsg::after {
  display: inline-block;
  position: relative;
  top: -2px;
  right: -12px;
  width: 10px;
  height: 10px;
  margin-left: -7px;
  content: '';
  background-color: #c8db16;
  border-radius: 50%;
}

.nav-links_container {
  display: block;
  width: 100vw;
  height: 0vh;
  -ms-grid-column: 1;
      grid-column-start: 1;
  grid-column-end: 7;
  background: #1c2324;
  opacity: 0;
  -webkit-transition: opacity 0.15s ease-out;
  transition: opacity 0.15s ease-out;
}

@media screen and (min-width: 1100px) {
  .nav-links_container {
    width: 100%;
    height: 100%;
    opacity: 1;
    -ms-grid-row: 2;
    grid-row: 2/2;
    -ms-grid-column: 5;
    -ms-grid-column-span: 2;
    grid-column: 5/7;
  }
}

.nav-links_container.open {
  opacity: 1;
  height: 100vh;
  z-index: 1;
}

@media screen and (min-width: 1100px) {
  .nav-links_container.open {
    height: unset;
  }
}

.nav-links {
  display: none;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  position: fixed;
  list-style: none;
  width: 70vw;
  height: 60vh;
  -webkit-transform: translate(15vw, 20vh);
          transform: translate(15vw, 20vh);
  -ms-flex-pack: distribute;
      justify-content: space-around;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  margin-left: auto;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
}

@media screen and (min-width: 1100px) {
  .nav-links {
    display: -ms-grid;
    display: grid;
    position: initial;
    -webkit-transform: none;
            transform: none;
    width: 100%;
    height: 100%;
    padding-top: 0px;
    -ms-grid-columns: (1fr)[4];
        grid-template-columns: repeat(4, 1fr);
  }
}

.nav-links .link {
  font-size: 25px;
  text-decoration: none;
  color: #e0e0e0;
}

@media screen and (min-width: 760px) {
  .nav-links .link {
    font-size: 35px;
  }
}

@media screen and (min-width: 900px) {
  .nav-links .link {
    font-size: 40px;
  }
}

@media screen and (min-width: 1100px) {
  .nav-links .link {
    margin-left: auto;
    margin-right: auto;
    font-size: 20px;
    z-index: 3;
  }
}

.nav-links .link:hover {
  color: #cede1a;
  cursor: pointer;
}

.nav-links .link:nth-child(1) {
  -webkit-transition: all 0.1s ease-in;
  transition: all 0.1s ease-in;
}

.nav-links .link:nth-child(2) {
  -webkit-transition: all 0.1s ease-in;
  transition: all 0.1s ease-in;
}

.nav-links .link:nth-child(3) {
  -webkit-transition: all 0.1s ease-in;
  transition: all 0.1s ease-in;
}

.nav-links.fade {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  opacity: 1;
}

@media screen and (min-width: 1100px) {
  .nav-links.fade {
    display: -ms-grid;
    display: grid;
  }
}
/*# sourceMappingURL=Nav.css.map */