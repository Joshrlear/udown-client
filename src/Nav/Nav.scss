@import "../utilities/_variables.sass";

@mixin desktop() {
  @media screen and ( min-width: 1024px ) {
    @content;
  }
}

@mixin tablet() {
  @media screen and ( min-width: 760px ) {
    @content;
  }
}

@mixin l-tablet() {
  @media screen and ( min-width: 900px ) {
    @content;
  }
}

* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
}

body {
  font-family: sans-serif;
}

nav {
  position: fixed;
  display: grid;
  grid-template-rows: 15% 70% 15%;
  grid-template-columns: repeat(6, 1fr);
  height: 8vh;
  width: 100vw;
  background: #1c2324;
  z-index: 1;
  @include l-tablet() {
    z-index: 5;
  }
}

  
.searchbar_container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  position: relative;
  grid-column-start: 1;
  grid-column-end: 5;
  grid-row-start: 2;
  grid-row-end: 2;
  @include desktop() {
    grid-column-end: 2; 
  }
}

.searchbar_input {
  border-radius: 20px;
  width: 35px;
  height: 35px;
  margin-top: 0px;
  margin-bottom: 0px;
  margin-right: 0px;
  margin-left: 20px;
  background-color: rgba(0,0,0,0);
  color: rgba(255, 255, 255, 0);
  padding-left: 15px;
  transition: all 0.2s ease-out;
}

.searchbar_input:focus {
  width: calc(100% - 20px);
  color: rgba(255, 255, 255, 0.6);
}

.searchbar_input::-webkit-input-placeholder {
  color: rgba(255, 255, 255, 0);
}

.searchbar_input:focus::-webkit-input-placeholder {
  color: rgba(255, 255, 255, 0.3);
  text-align: left;
}

#magnifying-glass {
  font-size: 40px; /* This controls the size. */
  display: inline-block;
  width: 0.4em;
  height: 0.4em;
  border: 2px solid rgba(255, 255, 255, 0.6);
  position: absolute;
  border-radius: 0.35em;
  margin-left: 28px;
  margin-top: -0.05vw;
  z-index: -1;
  transition: all 0.3s ease-out;
}

#magnifying-glass::before {
  content: "";
  display: inline-block;
  position: absolute;
  right: -6px;
  bottom: -3px;
  border-width: 0;
  background-image: linear-gradient(to right, rgba(255, 255, 255, 0.3), rgba(255, 255, 255, 1));
  width: 8px;
  height: 2px;
  -webkit-transform: rotate(45deg);
  -moz-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  -o-transform: rotate(45deg);
  transition: all 0.3s ease-out;
}

#magnifying-glass.off {
  border: none;
}

#magnifying-glass.off::before {
  background-image: linear-gradient(to right, rgba(255, 255, 255, 0), rgba(255, 255, 255, 0));
}

.search {
  display: flex;
  flex-direction: column;
  justify-content: center;
  position: relative;
  height: 100%;
  margin-left: 0px;
  grid-column-start: 6;
  grid-column-end: 6;
  grid-row-start: 2;
  grid-row-end: 2;
  z-index: 3;
  @include desktop() {
    grid-column: 2/2;
    margin-left: 20px;
  }
}

.search_btn_container {
  background: white;
  opacity: 0;
  position: relative;
  align-items: center;
  width: 35px;
  height: 35px;
  border-radius: 50%;
  display: grid;
  grid-template-rows: repeat(7, 1fr);
  grid-column-start: 6;
  grid-column-end: 6;
  grid-row-start: 4;
  grid-row-end: 5;
  z-index: 2;
  transition: opacity 0.35s ease-out;
  cursor: pointer;

  &.on {
    opacity: 1;
  }


  .search_btn {
    background: rgba(0, 0, 0, 0);
    height: 3px;
    width: 0px;
    grid-row-start: 4;
    grid-row-end: 5;
    z-index: -1;
    margin: 0;
    position: relative;
    cursor: pointer;
    transition: all 0.3s ease-out;

    &:before,
    &:after {
      content: "";
      background: rgba(0, 0, 0, 0);
      position: absolute;
      height: 3px;
      width: 15px;
      transition: all 0.3s ease-out;
    }

    &.on {
      width: 18px;
      border-top-right-radius: 25%;
      border-bottom-right-radius: 25%;
    }

    &.on, 
    &.on:before, 
    &.on:after {
      margin-left: 6px;
       background: #1c2324;
       z-index: 2;
    }

    &:before {
      right: -6.3px;
      bottom: -4px;
      transform: rotate(-45deg);
      border-radius: 3px;
    }

    &:after {
      right: -6.3px;
      top: -4px;
      transform: rotate(45deg);
      border-radius: 3px;
    }
  }
}

// to be integrated in next version
/* .newMsg_notification {
  display: none;
}

.newMsg_notification.newMsg {
  display: initial;
  background-color: $accent-color;
  border-radius: 50%;
  width: calc(1.7vw + 5px);
  height: calc(1.7vw + 5px);
  position: relative;
  top: 5%;
  right: 30%;
  margin-left: auto;
  margin-right: auto;
  grid-column-start: 6;
  grid-column-end: 7;
  grid-row-start: 2;
  grid-row-end: 2;
  pointer-events: none;
  z-index: 4;
  @include tablet() {
    right: 18%;
  }
  @include desktop() {
    display: none;
  }
} */

.hamburger_container {
  position: relative;
  left: 90%;
  display: grid;
  grid-template-rows: repeat(7, 1fr);
  grid-column-start: 5;
  grid-column-end: 6;
  grid-row-start: 2;
  grid-row-end: 2;
  z-index: 3;
  @include desktop() {
    display: none;
  }

  &.on {
    display: none;
    z-index: -1;
  }
}

.hamburger {
  cursor: pointer;
  grid-row: 2/3;
  grid-column: 2/6;
  @include tablet() {
    grid-column: 2/4;
    width: 85%;
  }
}

.line {
  height: 0.5vh;
  background: #e0e0e0;
  margin: 14%;
  transition: all 0.3s ease-out;


  &.on {
    width: 0px;
    margin-left: 35px;
    background: transparent;
    z-index: -1;
  }

}

// to be integrated in next version
/* .link.newMsg::after {
  display: inline-block;
  position: relative;
  top: -2px;
  right: -12px;
  width: 10px;
  height: 10px;
  margin-left: -7px;
  content: '';
  background-color: $accent-color;
  border-radius: 50%;
} */

.nav-links_container {
  display: block;
  width: 100vw;
  height: 0vh;
  grid-column-start: 1;
  grid-column-end: 7;
  background: #1c2324;
  opacity: 0;
  transition: opacity 0.15s ease-out;
  @include desktop() {
    width: 100%;
    height: 100%;
    opacity: 1;
    grid-row: 2/2;
    grid-column: 4/7;
    padding-left: 15%;
  }
}

.nav-links_container.open {
  opacity: 1;
  height: 100vh;
  z-index: 1;
  @include desktop() {
    height: unset;
  }
}

.nav-links {
  display: none;
  flex-direction: column;
  position: fixed;
  list-style: none;
  width: 70vw;
  height: 60vh;
  transform: translate(15vw, 20vh);
  justify-content: space-around;
  align-items: center;
  margin-left: auto;
  transition: all 0.25s;
  @include desktop() {
    display: grid;
    position: initial;
    transform: none;
    width: 100%;
    height: 100%;
    padding-top: 0px;
    grid-template-columns: repeat(4, 1fr);
  }
}

.nav-links .link{
  font-size: 25px;
  text-decoration: none;
  color: #e0e0e0;
  @include tablet() {
    font-size: 35px;
  }
  @include l-tablet() {
    font-size: 40px;
  }
  @include desktop() {
    margin-left: 10%;
    font-size: 20px;
    z-index: 3;
  }
}

.nav-links .link:hover {
  color: #cede1a;
  cursor: pointer;
}

.nav-links .link:nth-child(1) {
  transition: all 0.1s ease-in;
}

.nav-links .link:nth-child(2) {
  transition: all 0.1s ease-in;
}

.nav-links .link:nth-child(3) {
  transition: all 0.1s ease-in;
}

.nav-links.fade {
  display: flex;
  opacity: 1;
  @include desktop() {
    display: grid;
  }
}