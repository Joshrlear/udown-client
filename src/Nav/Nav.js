import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import UdownContext from '../UdownContext'
import functions from '../functions';
import './Nav.scss';

const authFunctions = functions.authFunctions

class Nav extends Component {
constructor(props) {
    super(props);
    this.queryRef = React.createRef()
    this.state = {
        toggledClass: false,
        searchbar_status: 'on',
        query: '',
        hasMsg: false,
    }
}


    static contextType = UdownContext

    componentDidUpdate() {
        const { hasMessage } = this.context
        hasMessage !== this.state.hasMsg 
            && (
                this.setState({
                    hasMsg: hasMessage
                })
            )
    }

    toggleMenu = e => {
        const { openMsg } =this.context
        if (window.innerWidth < 1099) {
            this.context.expandNav()
            this.setState({
                toggledClass: !this.state.toggledClass
            })
            e.target.innerHTML === 'Chat' && openMsg()
        }
    };

    viewChat = e => {
        const { isLoggedIn, openMsg } =this.context
        
        // close expanded nav menu
        this.context.expandNav()
            this.setState({
                toggledClass: !this.state.toggledClass
            })
        
        if (isLoggedIn) {
            // open chat
            e.target.innerHTML === 'Chat' && openMsg()
            this.state.hasMsg === true && (
                this.setState({
                    hasMsg: false
                })
            )  
        }

    }

    loginLogout = e => {
        if (this.context.isLoggedIn) {
            authFunctions.logout(this.props)
            this.context.setIsLoggedIn()
        }
        this.toggleMenu(e) 
    }

    inputFocus = () => {
        this.setState({
            searchbar_status: 'off'
        })
    }

    inputFocusout = () => {
        !this.state.query && (
            this.setState({ 
                searchbar_status: 'on' 
            })
        )
    }

    handleValue = e => {
        this.setState({
            query: this.queryRef.current.value
        })
    }

    searchQuery = e => {
        if (!e.key) {
            this.context.defineQuery(this.state.query)
            this.queryRef.current.value = ''
            const queryReset = Promise.resolve(this.setState({ query: '' }))
            queryReset.then(() => this.inputFocusout())
            this.props.history.push('/home')
        }
        else if (e.key === "Enter") {
            this.context.defineQuery(this.state.query)
            this.queryRef.current.value = ''
            this.setState({ query: '' })
            this.props.history.push('/home')
        }
    }

    render() {

        const navLinkClass = this.state.toggledClass ? "nav-links_container open" : "nav-links_container";
        const linkClass = this.state.toggledClass ? "nav-links fade" : "nav-links";
        const search_btn = !this.state.query ? "" : " on"
        const hasMsg = this.state.hasMsg ? "newMsg" : ""
        const loginLogout = this.context.isLoggedIn ? "Logout" : "Login"

        return (
            <nav>
            <div className="searchbar_container" onFocus={this.inputFocus} onBlur={this.inputFocusout}>
                <div id="magnifying-glass" className={this.state.searchbar_status}></div>
                    <input 
                        ref={this.queryRef} 
                        className="searchbar_input" 
                        placeholder="search..." 
                        onChange={this.handleValue} 
                        onKeyPress={this.searchQuery}
                    />
                </div>
                <div className="search" onClick={e => this.searchQuery(e)}>
                    <div className={`search_btn_container${search_btn}`}>
                        <div className={`search_btn${search_btn}`}>
                            <span className=""></span>
                        </div>
                    </div>
                </div>
                {/* to be integrated on next version */}
                {/* <div className={ `newMsg_notification ${hasMsg}` }/> */}
                <div className={`hamburger_container${search_btn}`}>
                    <div id="hamburger" className="hamburger" onClick={e => this.toggleMenu(e)}>
                      <div className={`line${search_btn}`}/>
                      <div className={`line${search_btn}`}/>
                      <div className={`line${search_btn}`}/>
                    </div>
                </div>
                <div className={ navLinkClass }>
                  <ul className={ linkClass }>
                    <NavLink className="link" name="home" to="/home" onClick={e => this.toggleMenu(e)}>Home</NavLink>
                    <span className={ `link ${hasMsg}`} name="chat" onClick={e => this.viewChat(e)}>Chat</span>
                    <NavLink className="link" name="profile" to="/profile" onClick={e => this.toggleMenu(e)}>Profile</NavLink>
                    <NavLink className="link" name="loginLogout" to="/login" onClick={e => this.loginLogout(e)}>{ loginLogout }</NavLink>
                  </ul>
                </div>
            </nav>
        );
    }
}

export default Nav;