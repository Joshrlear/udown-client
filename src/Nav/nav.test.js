import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16';

import Nav from './Nav';

configure({ adapter: new Adapter() })

describe('Nav', () => {
    it(`renders Nav without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(
        <BrowserRouter>
           <Nav/>
        </BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders Nav correctly', () => {
        const wrapper = shallow(<Nav/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})