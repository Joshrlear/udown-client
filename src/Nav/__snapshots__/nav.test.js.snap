// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders Nav correctly 1`] = `
<nav>
  <div
    className="searchbar_container"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <div
      className="on"
      id="magnifying-glass"
    />
    <input
      className="searchbar_input"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="search..."
    />
  </div>
  <div
    className="search"
    onClick={[Function]}
  >
    <div
      className="search_btn_container"
    >
      <div
        className="search_btn"
      >
        <span
          className=""
        />
      </div>
    </div>
  </div>
  <div
    className="hamburger_container"
  >
    <div
      className="hamburger"
      id="hamburger"
      onClick={[Function]}
    >
      <div
        className="line"
      />
      <div
        className="line"
      />
      <div
        className="line"
      />
    </div>
  </div>
  <div
    className="nav-links_container"
  >
    <ul
      className="nav-links"
    >
      <NavLink
        className="link"
        name="home"
        onClick={[Function]}
        to="/home"
      >
        Home
      </NavLink>
      <span
        className="link "
        name="chat"
        onClick={[Function]}
      >
        Chat
      </span>
      <NavLink
        className="link"
        name="profile"
        onClick={[Function]}
        to="/profile"
      >
        Profile
      </NavLink>
      <NavLink
        className="link"
        name="logout"
        onClick={[Function]}
        to="/login"
      >
        Login
      </NavLink>
    </ul>
  </div>
</nav>
`;
